test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Caching/CachingTestContext.cs

```diff
@@ -315,6 +315,7 @@ public void WriteProject()
             using (var fileStream = new FileStream(packagePath, FileMode.Open, FileAccess.Read))
             {
                 using (await GlobalPackagesFolderUtility.AddPackageAsync(
+                    string.Empty,
```

Comment:
`null`.  Throughout this PR pass `null` instead of `string.Empty` if there is no source.